src/common/exif.cc

```diff
@@ -393,6 +393,12 @@ class Lock
   image->readMetadata();                                      \
 }
 
+#define write_metadata_threadsafe(image)                      \
```

Comment:
Can we please move away from preprocessor macros, this so so 1990 style of writing C code :/ 
A local function (static or within an anonymous namespace) should do the job and if you don't trust the compiler to inline the code, you can even explicitly inline it.

Reply:
I don't have a strong opinion on that, I just took over the code from the read function. I guess they should be changed together, so I'm not sure if this is a task for this PR, and I don't know the dt style well enough to provide a well-formed commit.

Reply:
Let's fix one thing at a time.  That way we only have one thing to test at a time.  If someone wants to clean up the preprocessor macros, then let's do that in a separate PR and then we just have to test for regressions.